src/all/cosplaytele/src/eu/kanade/tachiyomi/extension/all/cosplaytele/CosplayTele.kt

```diff
@@ -0,0 +1,227 @@
+    override val lang = "all"
+    override val name = "CosplayTele"
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    // Latest
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        val manga = SManga.create()
+        manga.thumbnail_url = element.selectFirst("img")!!.attr("src")
+        val linkEl = element.selectFirst("h5 a")!!
+        manga.title = linkEl.text()
+        manga.setUrlWithoutDomain(linkEl.attr("abs:href"))
+        return manga
+    }
+
+    override fun latestUpdatesNextPageSelector() = ".next.page-number"
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/page/$page/")
+
+    override fun latestUpdatesSelector() = "div.box"
+
+    // Popular
+    override fun popularMangaFromElement(element: Element): SManga {
+        throw UnsupportedOperationException()
+    }
+
+    override fun popularMangaNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    private val popularPageLimit = 20
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/wp-json/wordpress-popular-posts/v1/popular-posts?offset=${page * popularPageLimit}&limit=$popularPageLimit&range=last7days")
+    override fun popularMangaSelector(): String = ""
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val jsonObject = json.decodeFromString<JsonArray>(response.body.string())
+        val mangas = jsonObject.map { item ->
+            val head = item.jsonObject["yoast_head_json"]!!.jsonObject
+            SManga.create().apply {
+                title = head["og_title"]!!.jsonPrimitive.content
+                thumbnail_url = head["og_image"]!!.jsonArray[0].jsonObject["url"]!!.jsonPrimitive.content
+                setUrlWithoutDomain(head["og_url"]!!.jsonPrimitive.content)
+            }
+        }
+        return MangasPage(mangas, mangas.size >= popularPageLimit)
+    }
+
+    // Search
+    override fun searchMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+    override fun searchMangaNextPageSelector() = latestUpdatesNextPageSelector()
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+
+        val categoryFilter = filterList.findInstance<UriPartFilter>()
+        return when {
+            categoryFilter?.state != 0 -> GET(
+                baseUrl.toHttpUrl().newBuilder().apply {
+                    addPathSegments(categoryFilter!!.toUriPart())
+                    addPathSegment("page")
+                    addPathSegment(page.toString())
+                    query.isNotEmpty().also {
+                        addQueryParameter("s", query)
+                    }
+                }.build(),
+            )
+            query.isNotEmpty() -> GET(
```

Comment:
move query search to top, so it gets prefered over filters

Reply:
I just copied how it is in Everia.Club

This so query can be used together with filter... If you filter first and use query, it works... But if query comes first it won't respect filter

Reply:
Here [CosplayTele.kt#L97](https://github.com/keiyoushi/extensions-source/blob/757c28cabcc95da3dbb5c5c53ba5d131326fc979/src/all/cosplaytele/src/eu/kanade/tachiyomi/extension/all/cosplaytele/CosplayTele.kt#L97)

Reply:
ah didn't see that